src/audio/pipeline.c

```diff
@@ -59,9 +59,45 @@ struct op_data {
 
 static void pipeline_task(void *arg);
 
+static void set_frames(struct pipeline *p, struct comp_dev *current,
```

Comment:
Would it not be better to walk the graph here from source to sink and have each component set it's number of frames (this would be based on rate, scheduling tick and source frames count).

Reply:
sorry didn't get this. As I understand it, this is what is exactly happening when set_frames is called from op_downstream/op_upstream, which are walking through the graph from start to end...?